### PR TITLE
Update manual testing for DG

### PR DESCRIPTION
**Pull request by NgChunMan**
- Improve the manual testing part for the Developer Guide.
- Add  json files to sample folder.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1077,33 +1077,408 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Shows the list of children in specified directory
+
+    1. Condition 1: When the current directory is root directory
+
+        - Test case: `ls`<br>
+          Expected: All the groups under root directory will be displayed.
+
+        - Test case: `ls grp-001`<br>
+          Expected: If there is no such a group with GroupId, `grp-001`, then an error message will appear at the output box below the command box. Otherwise, all the students in `grp-001` will be displayed.
```
Comment (mingyuanc) [inline docs/DeveloperGuide.md:1429]: Yeap something like this for the above one on CD

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1077,33 +1077,408 @@ testers are expected to do more *exploratory* testing.
...
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+    1. Prerequisites: Able to access the `/data` folder (created in the same folder containing the `ProfBook.jar` file).
 
-1. _{ more test cases …​ }_
+    1. Test case: Navigate to the `/data` folder and delete the `profbook.json` file. Then reopen the application.
```
Comment (mingyuanc) [inline docs/DeveloperGuide.md:1480]: I know this is what the other team did but follow the other one, ie the screenshot i send in the telegram group. IE the one where they make some changes using commands and it is reflected in the data file

**Review by mingyuanc — CHANGES_REQUESTED** [review]
Do check that the whole file do not repeat the changes that i requested. Other than that, make the changes and push it again, Dont resolve the changes, I will do so myself

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1157,33 +1157,450 @@ testers are expected to do more *exploratory* testing.
...
+              Expected: An error message of `Encountered invalid path element: x` will be shown.
 
-    1. Test case: `delete 0`<br>
-       Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
+        - Other incorrect `mkdir` commands to try: `mkdir`, `mkdir x`, `mkdir grp-001`, `mkdir --name Amazing Group1` (where x is an invalid GroupId)<br>
+              Expected: An error message of `Encountered invalid command format: missing argument(s)` will be shown.
 
-    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
-       Expected: Similar to previous.
+   1. Condition 1: When the current directory is group directory,
```
Comment (mingyuanc) [inline docs/DeveloperGuide.md:1175]: This should be condition 2 ah

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1157,33 +1157,450 @@ testers are expected to do more *exploratory* testing.
...
+
+       - Other incorrect `rm` commands to try: `rm`<br>
+         Expected: An error message of `Encountered invalid command format: missing argument(s)` will be shown.
+
+   1. Condition 2: When the current directory is group directory e.g. `grp-001`,
+
+       - Test case: `rm 0123Y`<br>
+         Expected: If there is no such a student with StudentId `0123Y` in `grp-001`, then an error message will appear at the output box below the command box. Otherwise, the student `0123Y` in `grp-001` wil be deleted.
+
+       - Test case: `rm grp-001`<br>
```
Comment (mingyuanc) [inline docs/DeveloperGuide.md:1235]: image

Expected outcome wrong

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1077,33 +1077,408 @@ testers are expected to do more *exploratory* testing.
...
+         Expected: If there is no such a group with GroupId, `grp-001`, then an error message will appear at the output box below the command box. Otherwise, the group, `grp-001` wil be deleted.
+
+       - Test case: `rm x` (where x is an invalid path)<br>
+         Expected: An error message of `Invalid relative path` will be shown.
+      
+       - Other incorrect `rm` commands to try: `rm`<br>
+         Expected: An error message of `Invalid command format` will be shown.
+
+
+### Editing a student's details or group's details
```
[the remark this answers is not in the document]
Comment (mingyuanc) [reply docs/DeveloperGuide.md:1245]: You did not resolve this ah

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1077,33 +1077,408 @@ testers are expected to do more *exploratory* testing.
...
+
+       - Other incorrect `deadline` commands to try: `deadline`, `deadline x` (where x is an invalid path)<br>
+         Expected: An error message of `Invalid command format` will be shown.
+
+
+### Mark task as completed
+
+1. Marks the specified task as done for the specified student or group
+
+    1. Prerequisites: The display panel is showing the task list of the specific student or group.
```
[the remark this answers is not in the document]
Comment (mingyuanc) [reply docs/DeveloperGuide.md]: You did it for mark task but you did not do for unmark/delete tast

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1157,33 +1157,450 @@ testers are expected to do more *exploratory* testing.
...
+    1. Condition 1: When the current directory is root directory
+
+        - Test case: `ls`<br>
+          Expected: All the groups under root directory will be displayed.
+
+        - Test case: `ls grp-001`<br>
+          Expected: If there is no such a group with GroupId `grp-001`, then an error message will appear at the output box below the command box. Otherwise, the interface will change from displaying all the group to displaying all the student in `grp-001`.
+
+        - Test case: `ls x` (where x is an invalid path)<br>
+          Expected: An error message of `Encountered invalid path element: x will be shown.
```
Comment (mingyuanc) [inline docs/DeveloperGuide.md:1544]: your missing a back tick

**Review by mingyuanc — CHANGES_REQUESTED** [review]
Got some changes not done properly! Do not use numbered bullet points cause it looks super super weird

File: docs/DeveloperGuide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (NgChunMan) [reply docs/DeveloperGuide.md:1245]: Are these two lines (line 1257 and line 1277) not what you referring to?

**Review by mingyuanc — APPROVED** [review]
LGTM!!!!!!

**Pull request merged**